serialization/api.go

```diff
@@ -463,100 +463,244 @@ const (
 	ReadInt64(fieldName string) int64
+	// ReadFloat32 reads and returns a float32
 	ReadFloat32(fieldName string) float32
+	// ReadFloat64 reads and returns a float64
 	ReadFloat64(fieldName string) float64
+	// ReadString reads and returns a string which may be nil.
 	ReadString(fieldName string) *string
+	// ReadDecimal reads and returns a *types.Decimal value which may be nil.
 	ReadDecimal(fieldName string) *types.Decimal
+	// ReadTime reads and returns a *types.LocalTime value which may be nil.
 	ReadTime(fieldName string) *types.LocalTime
+	// ReadDate reads and returns a *types.LocalDate value which may be nil.
 	ReadDate(fieldName string) *types.LocalDate
+	// ReadTimestamp reads and returns a *types.LocalDateTime value which may be nil.
 	ReadTimestamp(fieldName string) *types.LocalDateTime
+	// ReadTimestampWithTimezone reads and returns a *types.OffsetDateTime value which may be nil.
 	ReadTimestampWithTimezone(fieldName string) *types.OffsetDateTime
+	// ReadCompact reads and returns a compact serialized value which may be nil.
 	ReadCompact(fieldName string) interface{}
+	// ReadArrayOfBoolean reads and returns a slice of bools.
 	ReadArrayOfBoolean(fieldName string) []bool
+	// ReadArrayOfInt8 reads and returns a slice of int8s.
 	ReadArrayOfInt8(fieldName string) []int8
+	// ReadArrayOfInt16 reads and returns a slice of int16s.
 	ReadArrayOfInt16(fieldName string) []int16
+	// ReadArrayOfInt32 reads and returns a slice of int32s.
 	ReadArrayOfInt32(fieldName string) []int32
+	// ReadArrayOfInt64 reads and returns a slice of int64s.
 	ReadArrayOfInt64(fieldName string) []int64
+	// ReadArrayOfFloat32 reads and returns a slice of float32s.
 	ReadArrayOfFloat32(fieldName string) []float32
+	// ReadArrayOfFloat64 reads and returns a slice of float64s.
 	ReadArrayOfFloat64(fieldName string) []float64
+	// ReadArrayOfString reads and returns a slice of string pointers.
+	// The returned pointers may be nil.
 	ReadArrayOfString(fieldName string) []*string
+	// ReadArrayOfDecimal reads and returns a slice of type.Decimal pointers.
+	// The returned pointers may be nil.
 	ReadArrayOfDecimal(fieldName string) []*types.Decimal
+	// ReadArrayOfTime reads and returns a slice of type.LocalTime pointers.
+	// The returned pointers may be nil.
 	ReadArrayOfTime(fieldName string) []*types.LocalTime
+	// ReadArrayOfDate reads and returns a slice of type.LocalDate pointers.
+	// The returned pointers may be nil.
 	ReadArrayOfDate(fieldName string) []*types.LocalDate
+	// ReadArrayOfTimestamp reads and returns a slice of type.LocalDateTime pointers.
+	// The returned pointers may be nil.
 	ReadArrayOfTimestamp(fieldName string) []*types.LocalDateTime
+	// ReadArrayOfTimestampWithTimezone reads and returns a slice of type.OffsetDateTime pointers.
+	// The returned pointers may be nil.
 	ReadArrayOfTimestampWithTimezone(fieldName string) []*types.OffsetDateTime
+	// ReadArrayOfCompact reads and returns a slice of values.
```

Comment:
Is there any chance it could be nil?

Reply:
The slice can be nil, but it is treated as empty slice by Go.

Reply:
The values themselves may be nil.